### DISCUSSION TITLE
Not supplying a config level context and using assign should be an error

### DISCUSSION THREAD

**Discussion by UberMouse** [statelyai/xstate#1227] [General] [answered]
I just got bit by this, probably spent approx 7h debugging everything under the sun I could think of. Just didn't think to check I had set an initial context for the machine in the config (because my initial context was an empty object).

I guess I had assumed that xstate would create an empty context object if I tried to use `assign` to set something on the context. I was lead in the wrong direction when I couldn't see the `assign` action in the `actions` property of the state object I was logging in `onTransition`. Thought somehow the `assign` was getting removed from the `entry` actions.

It feels like this should be an error instead of xstate just silently failing to assign new values to the context object.

**Comment by davidkpiano** [comment, marked as answer]
Can you move this to issues so we can fix it?

**Reply by UberMouse** [reply]
Ah, didn't even realize this wasn't in issues. I just clicked feature request. I'll do that now.
